app/src/main/java/ch/admin/bag/dp3t/networking/FakeWorker.java

```diff
@@ -32,27 +33,36 @@
 	private static final String WORK_TAG = "ch.admin.bag.dp3t.FakeWorker";
 	private static final String FAKE_AUTH_CODE = "000000000000";
 
+	private static final long FACTOR_HOUR_MILLIS = 60 * 60 * 1000L;
+	private static final long FACTOR_DAY_MILLIS = 24 * FACTOR_HOUR_MILLIS;
+	private static final long MAX_DELAY_HOURS = 48;
 	private static final float SAMPLING_RATE = BuildConfig.FLAVOR.equals("dev") ? 1.0f : 0.2f;
-	private static final long FACTOR_DAY_MILLIS = 24 * 60 * 60 * 1000L;
+	private static final String KEY_T_DUMMY = "KEY_T_DUMMY";
 
 	public static void safeStartFakeWorker(Context context) {
-		startFakeWorker(context, ExistingWorkPolicy.KEEP);
+		long t_dummy = SecureStorage.getInstance(context).getTDummy();
```

Comment:
Mhhhhm, the initial value (on app initialization)  is not written to SecureStorage until the worker actually runs. Could that cause several `t_dummy` times to be sampled? All of which will be scheduled?

Answering my own  question: no, `ExistingWorkPolicy.KEEP` ensures that it is not added again, and the first sample will persist. Of course, this assumes that workers are properly "archived" and restarted by Android.

What is the function of SecureStorage of TDummy if these values are always pulled and updated from `t_dummy` passed to the worker?

Reply:
Yes, in theory it should be ok because the WorkJobs should not be lost. But to not have to completely trust this system (some batterysaver-mode removing our jobs) we store it anyway, so if we should be freshly scheduled at some point, the behaviour does not change. But with this argument, we definitely should also store the initial t_dummy!

Reply:
Cool!